src/pulumi_datarobot_utils/schema/exec_envs.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2025 DataRobot, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from enum import Enum
+
+import datarobot as dr
+
+from pulumi_datarobot_utils.schema.base import Schema
+
+
+class RuntimeEnvironment(Schema):
+    name: str
+
+    @property
+    def id(self) -> str:
+        client = dr.client.get_client()
+        try:
+            environments = client.get(
+                "executionEnvironments/", params={"searchFor": self.name}
```

Comment:
So this immediately stands out to me as either:
1. Something that is missing from the [DR Python Client](https://pypi.org/project/datarobot/) and we should add or make sure the usage needed is supported _or_ 
2. It _does_ exist in the DR Python Client and we should use that

In this specific case I'm _pretty_ sure we could simplify this code by using:
`ExecutionEnvironment::list()`

Docs here:
https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/custom_models.html#datarobot.ExecutionEnvironment.list

It has the parameter `search_for` so I would hope it already has functionality that's required.

Reply:
Great idea 👍 
Left a todo to consider this: https://github.com/datarobot-oss/pulumi-datarobot-utils/pull/1/commits/2f8dde63472a89b4846ee3fb7590d82fd4e4fa0d

Thank you!